code/game/machinery/rechargestation.dm

```diff
@@ -47,6 +47,12 @@
 	if(!isrobot(occupant) && !ishuman(occupant))
 		return
 
+	var/obj/item/cell/cell = get_cell()
+	if(!cell || (cell.charge <= 0))
+		go_out()
+		playsound(src.loc, 'sound/machines/buzz-sigh.ogg', 50, 1)
```

Comment:
Probably for the best to add a visible message so someone isn't confused why they were suddenly ejected.